examples/linkedlist.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import print_function, absolute_import
+from numba.utils import OrderedDict
+import numpy as np
+from numba import njit
+from numba.jitclass import jitclass
+from numba import int32, deferred_type, optional
+from numba.runtime import rtsys
+
+node_type = deferred_type()
+
+spec = OrderedDict()
+spec['data'] = int32
+spec['next'] = optional(node_type)
+
+
+@jitclass(spec)
+class LinkedNode(object):
+    def __init__(self, data, next):
+        self.data = data
+        self.next = next
+
+    def prepend(self, data):
+        return LinkedNode(data, self)
+
+
+@njit
+def make_linked_node(data):
+    return LinkedNode(data, None)
+
+
+node_type.define(LinkedNode.class_type.instance_type)
```

Comment:
Ok, I have tried implement the ``deferred(local_name)`` approach and I think it is more complicated then the current approach.  You can find my failed attempt at https://github.com/sklam/numba/compare/experiment/jitclasses...experiment/jitclasses_alt . I can get one of the example working but not the tests.  Resolving the local name later is harder than I thought.  For some reason, frame.f_locals may not have what I need.  A major source of complexity is at "resolving" the Type while they are expected to be immutable.